server/src/storage/uncompressed_data_manager.rs

```diff
@@ -47,23 +54,172 @@ impl UncompressedDataManager {
+        data_points: &RecordBatch,
+    ) -> Result<(), String> {
+        let _span = info_span!("insert_data_points", table = model_table.name).entered();
+        info!(
+            "Received record batch with {} data points for the table '{}'.",
+            data_points.num_rows(),
+            model_table.name
+        );
+
+        // Prepare the timestamp column for iteration.
+        let timestamp_index = model_table.timestamp_column_index as usize;
+        let timestamps: &TimestampArray = data_points
+            .column(timestamp_index)
+            .as_any()
+            .downcast_ref()
+            .unwrap();
+
+        // Prepare the tag columns for iteration.
+        let tag_column_arrays: Vec<&StringArray> = model_table
+            .tag_column_indices
+            .iter()
+            .map(|index| {
+                data_points.column(*index as usize)
+                    .as_any()
+                    .downcast_ref()
+                    .unwrap()
+            })
+            .collect();
+
+        // Prepare the field columns for iteration. The column index is saved with the corresponding array.
+        let field_column_arrays: Vec<(usize, &ValueArray)> = model_table
+            .schema
+            .fields()
+            .iter()
+            .filter_map(|field| {
+                let index = model_table.schema.index_of(field.name().as_str()).unwrap();
+
+                // Field columns are the columns that are not the timestamp column or one of the tag columns.
+                let not_timestamp_column = index != model_table.timestamp_column_index as usize;
+                let not_tag_column = !model_table.tag_column_indices.contains(&(index as u8));
+
+                if not_timestamp_column && not_tag_column {
+                    Some((
+                        index,
+                        data_points.column(index as usize).as_any().downcast_ref().unwrap(),
+                    ))
+                } else {
+                    None
+                }
+            })
+            .collect();
+
+        // For each row in the data, generate a tag hash, extract the individual measurements,
+        // and insert them into the storage engine.
+        for (index, timestamp) in timestamps.iter().enumerate() {
+            let tag_values: Vec<String> = tag_column_arrays
+                .iter()
+                .map(|array| array.value(index).to_string())
+                .collect();
+
+            let tag_hash = self
+                .get_or_compute_tag_hash(&model_table, &tag_values)
+                .map_err(|error| format!("Tag hash could not be saved: {}", error.to_string()))?;
+
+            // For each field column, generate the 64-bit key, create a single data point, and
+            // insert the data point into the in-memory buffer.
+            for (field_index, field_column_array) in &field_column_arrays {
+                let key = tag_hash | *field_index as u64;
```

Comment:
should we have a "typedef"/alias for this? Or is that an overkill?

Reply:
I think it might be overkill especially considering what is also mentioned in [this](https://github.com/ModelarData/ModelarDB-RS/pull/30#discussion_r973097778) answer.